packages/compass-saved-aggregations-queries/src/components/open-item-modal.tsx

```diff
@@ -134,11 +140,12 @@ const OpenItemModal: React.FunctionComponent<OpenItemModalProps> = ({
   onClose,
   onSubmit,
 }) => {
+  const [updateNamespace, setUpdateNamespace] = useState(false);
```

Comment:
One struggle with leafygreen modals that we always have is resetting the state after the modal is closed. Because the `OpenItemModal` is never unmounted, the modal is just closed with the prop, this `updateNamespace` state is never reset, so next time you open the modal, the checkbox is enabled by default.

You have two ways to deal with this here. You can either move this state to the reducer where the selected namespace state exists and reset the value on modal close with the other logic that resets the state. It's a bit more involved, but has the benefit of conststency: all form state is handled in one place and with one mechanism. Another way of dealing with this is adding an effect here in UI code that will set the value back to false when the modal is closed. This is simpler, but means that this form state is now handled in multiple places making it a bit harder to maintain.